common/src/main/java/com/wynntils/models/spells/SpellModel.java

```diff
@@ -60,14 +64,21 @@ public void onItemRenamed(ItemRenamedEvent event) {
         }
     }
 
+    @SubscribeEvent
+    public void onWorldChange(WorldStateEvent e) {
+        lastSpell = SpellDirection.NO_SPELL;
+        lastSpellUpdate = Instant.EPOCH;
+    }
+
     @SubscribeEvent
     public void onSpellSegmentUpdate(SpellSegmentUpdateEvent e) {
         Matcher matcher = e.getMatcher();
         if (!matcher.matches()) return;
 
         SpellDirection[] spell = getSpellFromMatcher(e.getMatcher());
-        if (Arrays.equals(spell, lastSpell)) return; // Wynn sometimes sends duplicate packets, skip those
-        lastSpell = spell;
+        if (Duration.between(lastSpellUpdate, Instant.now()).toSeconds() < 3 && Arrays.equals(spell, lastSpell))
+            return; // Wynn sometimes sends duplicate packets, skip those
```

Comment:
Comment is out-of-date and there is a multiline if without braces..

Reply:
so it doesn't send multiple packets anymore?